Gem/Code/Source/Automation/ScriptReporter.cpp

```diff
@@ -1081,4 +1091,46 @@ namespace AtomSampleViewer
         }
 
     }
+
+    void ScriptReporter::ExportTestResults()
+    {
+        uint32_t totalAssertCount = 0;
+        uint32_t totalErrorCount = 0;
+        uint32_t totalWarningCount = 0;
+        uint32_t totalScreenshotErrorCount = 0;
+        uint32_t totalScreenshotWarningCount = 0;
+        
+        for (ScriptReport& scriptReport : m_scriptReports)
+        {
+            totalAssertCount += scriptReport.m_assertCount;
+            totalErrorCount += scriptReport.m_generalErrorCount;
+            totalWarningCount += scriptReport.m_generalWarningCount;
+            totalScreenshotErrorCount += scriptReport.m_screenshotErrorCount;
+            totalScreenshotWarningCount += scriptReport.m_screenshotWarningCount;
+        }
+
+        AZStd::string assertLogLine = AZStd::string::format("Asserts: %i \n", totalAssertCount);
+        AZStd::string errorsLogLine = AZStd::string::format("Errors: %i \n", totalErrorCount);
+        AZStd::string warningsLogLine = AZStd::string::format("Warnings: %i \n", totalWarningCount);
+        AZStd::string screenshotErrorsLogLine = AZStd::string::format("Screenshot errors: %i \n", totalScreenshotErrorCount);
+        AZStd::string screenshotWarningsLogLine = AZStd::string::format("Screenshot warnings: %i \n", totalScreenshotWarningCount);
+
+        auto io = AZ::IO::LocalFileIO::GetInstance();
+        AZStd::string projectPath = AZ::Utils::GetProjectPath();
+        AZStd::string exportFileName = "exportedTestResults.txt";
+        AzFramework::StringFunc::Path::Join(projectPath.c_str(), "TestResults/", m_exportTestResultsFolder);
+        AzFramework::StringFunc::Path::Join(m_exportTestResultsFolder.c_str(), "exportedTestResults.txt", exportFileName);
+        AZ::IO::HandleType logHandle;
+        
+        io->CreatePath(m_exportTestResultsFolder.c_str());
+        io->Open(exportFileName.c_str(), AZ::IO::OpenMode::ModeWrite, logHandle);
+        io->Write(logHandle, assertLogLine.c_str(), assertLogLine.size());
+        io->Write(logHandle, errorsLogLine.c_str(), errorsLogLine.size());
+        io->Write(logHandle, warningsLogLine.c_str(), warningsLogLine.size());
+        io->Write(logHandle, screenshotErrorsLogLine.c_str(), screenshotErrorsLogLine.size());
+        io->Write(logHandle, screenshotWarningsLogLine.c_str(), screenshotWarningsLogLine.size());
+        io->Close(logHandle);
```

Comment:
I think we will also need to write the individual test reports into this file. That should be the last piece we will need to see in the .txt file. I think it would be a good idea to format the text file too, add \n to separate each tests, and use \t to add tabs appropriately.